globals/matchmaking_globals.go

```diff
@@ -0,0 +1,12 @@
+package common_globals
+import (
+	match_making "github.com/PretendoNetwork/nex-protocols-go/match-making"
+)
+
+type CommonMatchmakeSession struct {
+	GameMatchmakeSession match_making.MatchmakeSession //used by the game, contains the current state of the MatchmakeSession
+	SearchMatchmakeSession match_making.MatchmakeSession //used by the server when searching for matches, contains the state of the MatchmakeSession during the search process for easy compares
+	PlayersByConnectionId []uint32 //players in the room, referenced by their connection IDs. This is used instead of the PID in order to ensure we're talking to the correct client (in case of e.g. multiple logins)
```

Comment:
In Go the style guidelines dictate that identifiers like `ConnectionId` should actually be `ConnectionID` (this also applies to other names like `HTTP` and `URL`) https://github.com/golang/go/wiki/CodeReviewComments#initialisms

The name `PlayersByConnectionID`, while descriptive and correct, does feel a bit long-winded and awkward. Not exactly an issue and it can stay as is, since it is correct, but maybe something shorter like `UserIDs` or `ConnectionIDs` would be better?

Or alternatively, maybe we should store this based on the users PID, and then reference that client through the NEX server by the PID? I am unsure if the users connection ID would ever change but if it does then that would become an issue if we only store the connection ID. Again not an issue right now, just some food for thought. If it stays that's also fine :)